### PR TITLE
[RW-7112][risk=low] rm unused Elasticsearch integration

### PR DESCRIPTION
**Pull request by calbach**
Given efforts to pull out / generalize dataset builder functionality, and that Elastic integration is still not approved, it is highly unlikely we will use this integration in the current version of Workbench. Of course, we can just revert this PR if those plans change.

### CODE REVIEW

File: api/src/main/java/org/pmiops/workbench/cohortbuilder/util/CriteriaLookupUtil.java [hunk 1]
```diff
@@ -27,10 +27,7 @@
 
 /**
  * This lookup utility extracts all criteria groups in the given search request and produces a
- * lookup map from group parameter to set of matching child concept ids. This class was implemented
- * to consolidate logic between the CB query builders and elastic search filters. The ElasticFilters
- * class will need to be refactored to use this lookup util going forward:
- * TODO:https://precisionmedicineinitiative.atlassian.net/browse/RW-2875
+ * lookup map from group parameter to set of matching child concept ids.
  */
 public final class CriteriaLookupUtil {
```
Comment (freemabd) [inline api/src/main/java/org/pmiops/workbench/cohortbuilder/util/CriteriaLookupUtil.java:32]: This class was used exclusively by the Elastic implementation. Should we remove it as well? 
Also, in `CBCriteriaDao` we have 4 methods that are exclusive to  `CriteriaLookupUtil`
`findCriteriaAncestors`
`findCriteriaParentsByDomainAndTypeAndParentConceptIds`
`findCriteriaLeavesAndParentsByDomainAndTypeAndParentIds`
`findCriteriaLeavesAndParentsByPath`

**Review by freemabd — APPROVED** [review]
Seems BQ integration tests are failing with org.hibernate.LazyInitializationException. I've not seen this before. I'm assuming this is not a side effect of this PR.

File: api/src/main/java/org/pmiops/workbench/cohortbuilder/util/CriteriaLookupUtil.java (hunk 1 above)
Comment (calbach) [reply api/src/main/java/org/pmiops/workbench/cohortbuilder/util/CriteriaLookupUtil.java]: Thanks - missed that. Removed.

**Comment by calbach** [comment]
I believe https://github.com/all-of-us/workbench/pull/5314 should have fixed - I may have failed to rebase before sending this PR.

**Pull request merged**